### PR TITLE
Feature/instructor calendar

### PR DESCRIPTION
**Pull request by JoeyMBrown**
What have I done?
- Created folders for InstructorBooking, and Parent booking along with index.js files within them.
- Added a course details card component from Ant Design that could be used with either calendar (currently not important anywhere)
- Converted InstructorCalendar to use Redux instead of local state.
- Changed import for less file inside of InstructorProfile.
- Removed import for InstructorProfile inside of InstructorHome
- Inside of ParentHome removed duplicate calendar.  Added back Col spacing to add "margins" to right side of calendar.
- Removed main footer from index.js as there is temporarily a bug with it that breaks the styling for sidebar.
- Instructor actions and instructor reducer edited to change calendar to redux
- Added a md file of a potential style to use along with the course details component (currently this seems to be a duplicate component, opting for a conversation with FE to decide which we'd like to move forward with)
- Added a missing closing bracket to index.less file to make project compile properly.

What files were changed?
- Added InstructorBooking and ParentBooking folders, with empty index.js files
- InstructorCalendar.js
- InstructorHome.js
- ParentHome.js
- src/index.js
- InstructorActions.js, InstructorReducer.js
- StylesForPotentialComponent.md created - will be removed soon (created initially to avoid merge conflicts)

Why was this change made?
- InstructorBooking and ParentBooking folders were created to establish file scaffolding for future parentBooking and InstructorBooking pages
- CourseDetails component was created for a place to display course information from calendar on calendar click (may be duplicate component)
- Converting Instructor calendar to redux saves multiple lines of code from using local state.
- Import inside of instructorProfile was changed to prevent merge conflicts and to match other PR's
- InstructorProfile import was removed as it was no longer needed (doesn't match UX mockups)
- Due to merges ParentHome had 2 calendars being imported, 1 needed to be removed.
- Footer currently has a css bug that breaks sidebar css slightly, temporarily removed it until we find a fix.
- Changes to InstructorActions and InstructorsReducer was necessary to properly link up InstructorCalendar to Redux
- Missing bracket fix needed to make project compile properly.

[Loom Video](https://www.loom.com/share/d445ca7a136240f08f6adbac7944fe00)

co-authored-by: @Rodgers31

### CODE REVIEW

File: src/components/pages/InstructorHome/InstructorCalender.js [hunk 1]
```diff
@@ -1,39 +1,56 @@
...
   function getListData(value) {
-    const values = instructor.course_schedule.map(itemz => {
-      setCourse(itemz.subject);
+    let listData = [];
+
+    // Here we're mapping over the dates of each course.  We keep track of the index we're looking at currently.
+    // 1. If the current day matches value.date, it must mean this course is for the day we called getListData with.
+    // 2. If the days match, we need to update listData for that day.  We grab the courses subject from course state using the index.  Indexes will always match.
+    // 3. Important to note listdata is spread here to include the possibility of multiple courses occuring on one day.
+    instructor.course_schedule.forEach(currentCourse => {
```
Comment (MikeLikesCode) [inline src/components/pages/InstructorHome/InstructorCalender.js:17]: Awesome way of looping through the data

**Review by MikeLikesCode — APPROVED** [review]

**Review by chelseaceballos — APPROVED** [review]
All changes look good to me. Nice work changing InstructorCalendar to use Redux!

File: src/components/pages/InstructorHome/InstructorCalender.js [hunk 2]
```diff
@@ -1,39 +1,56 @@
...
 const InstructorCalender = props => {
   const { instructor } = props;
-  const [course, setCourse] = useState('');
 
   function getListData(value) {
-    const values = instructor.course_schedule.map(itemz => {
-      setCourse(itemz.subject);
+    let listData = [];
+
+    // Here we're mapping over the dates of each course.  We keep track of the index we're looking at currently.
```
Comment (ryan-hamblin) [inline src/components/pages/InstructorHome/InstructorCalender.js:13]: Says we're mapping but I see a for loop instead.

File: src/components/pages/InstructorHome/InstructorCalender.js [hunk 3]
```diff
@@ -1,39 +1,56 @@
...
+    // 1. If the current day matches value.date, it must mean this course is for the day we called getListData with.
+    // 2. If the days match, we need to update listData for that day.  We grab the courses subject from course state using the index.  Indexes will always match.
+    // 3. Important to note listdata is spread here to include the possibility of multiple courses occuring on one day.
+    instructor.course_schedule.forEach(currentCourse => {
+      if (
+        parseInt(currentCourse.start_date.split('-')[1], 10) === value.date() &&
+        value.month() + 1 ===
+          parseInt(currentCourse.start_date.split('-')[0], 10)
+      ) {
+        //Needs to be +1 because month indexing starts at 0
```
Comment (ryan-hamblin) [inline src/components/pages/InstructorHome/InstructorCalender.js:23]: Love that we're learning allll the quirks today!

File: src/components/pages/InstructorHome/InstructorCalender.js [hunk 4]
```diff
@@ -1,39 +1,56 @@
...
+      });
+    }
 
     return (
       <ul className="events">
         {listData.map(item => (
-          <li key={item.content}>
-            <Badge status={item.type} text={item.course} />
+          <li id={item.id} key={item.id} onClick={evt => showCourseInfo(evt)}>
+            {/** Need to find a better way to add onClick event.  Right now it's not properly attached to the text inside of calender.  It's slightly below it */}
```
Comment (ryan-hamblin) [inline src/components/pages/InstructorHome/InstructorCalender.js:52]: Let's get this bug documented in the Trello board if it's not already.

File: src/components/pages/InstructorHome/InstructorCalender.js [hunk 5]
```diff
@@ -43,6 +60,15 @@ const InstructorCalender = props => {
   return (
     <>
       <section className="site-calendar-demo-card">
+        <div>
+          <h1>Your Calender</h1>
+          <div className="sub-items">
+            {/* <h2>Subheader</h2> */}
```
Comment (ryan-hamblin) [inline src/components/pages/InstructorHome/InstructorCalender.js:66]: Please remove this if it's commented out.

File: src/index.js [hunk 6]
```diff
@@ -76,7 +76,7 @@ function App() {
         </div>
         <Route component={NotFoundPage} />
       </Switch>
-      <MainFooter />
+      {/*<MainFooter />*/}
```
Comment (ryan-hamblin) [inline src/index.js:79]: Please delete before it goes to main.

File: src/redux/reducers/instructorReducer.js [hunk 7]
```diff
@@ -4,9 +4,22 @@ import {
   GET_USER_ACTION,
   GET_COURSES_ACTION,
   GET_INBOX_ACTION,
+  SET_SELECTED_COURSE,
 } from '../actions/instructorActions';
 
 const initialState = {
```
Comment (ryan-hamblin) [inline src/redux/reducers/instructorReducer.js:10]: I personally like to put dummyData in a separate file.

**Review by muhannadbm — APPROVED** [review]
looks good

File: src/styles/InstructorStyles/StylesForPotentialComponent.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
+Export other less files here
```
Comment (ryan-hamblin) [inline src/styles/InstructorStyles/StylesForPotentialComponent.md:1]: Why was a MD file needed for the defining of `.less` styles?

**Review by ryan-hamblin — COMMENTED** [review]
Great work. There are a few items that I mentioned that I'd like you to pay attention to. 

Please remove the linter warnings, and the commented out code. 

Please consider the Date() constructor when comparing dates in lieu using `parseInt` to do so.

**Pull request merged**